scripts/qa/capture_model_row_counts.py

```
@@ -0,0 +1,391 @@
+
+def discover_models(
+    conn: snowflake.connector.SnowflakeConnection,
+    databases: list[str],
+) -> list[tuple[str, str, str, str]]:
+    """Return (db, schema, name, table_type) for every table/view in scope."""
+    parts = []
+    for db in databases:
+        parts.append(
+            f"SELECT '{db}' AS table_catalog, table_schema, table_name, table_type "
+            f"FROM {db}.INFORMATION_SCHEMA.TABLES "
+            f"WHERE table_type IN ('BASE TABLE', 'VIEW')"
```

Ruff (0.15.13)

[error] Possible SQL injection vector through string-based query construction

(S608)